### PR TITLE
[IDP-1457] Fix clippy issues

### PR DESCRIPTION
**Pull request by janslow**

### CODE REVIEW

File: shipcat_cli/src/helm.rs [hunk 1]
```diff
@@ -216,13 +215,7 @@ fn check_labels(mf: &Manifest, kind: &str, skipped: &[String], obj: &KubeObject)
 fn check_owner_refs(mf: &Manifest, kind: &str, obj: &KubeObject) -> Result<bool> {
     let mut success = true;
     // First ownerReferences must be ShipcatManifest
-    match obj
-        .metadata
-        .owner_references
-        .clone()
-        .unwrap_or_else(|| vec![])
```
Comment (janslow) [inline shipcat_cli/src/helm.rs:223]: `.unwrap_or_else(|| vec![])` -> `.unwrap_or_else(vec![])` -> `.unwrap_or_default()`

File: shipcat_cli/src/kong.rs [hunk 2]
```diff
@@ -36,7 +36,7 @@ impl KongfigOutput {
             host: data.kong.clone().config_url,
             headers: vec![],
             apis: kongfig_apis(data.apis, data.kong.clone(), region),
-            consumers: kongfig_consumers(data.kong.clone()),
+            consumers: kongfig_consumers(data.kong),
```
Comment (janslow) [inline shipcat_cli/src/kong.rs:39]: unnecessary clone

File: shipcat_definitions/src/crds.rs [hunk 3]
```diff
@@ -16,6 +16,7 @@ impl From<Manifest> for ShipcatManifest {
         // we assume the manifest has all it needs to fill in the pieces
         // but no secrets!
         assert_eq!(mf.state, ManifestState::Base);
+        #[allow(clippy::redundant_clone)] // cloning the name is cheaper than cloning the manifest
         ShipcatManifest::new(&mf.name.clone(), mf)
```
Comment (janslow) [inline shipcat_definitions/src/crds.rs:20]: Clippy was complaining about an unnecessary clone of `&mf.name`, but if we don't do that, we need to do:

```rust
         ShipcatManifest::new(&mf.name, mf.clone())
```

Which is more expensive

File: shipcat_definitions/src/structs/env.rs [hunk 4]
```diff
@@ -88,7 +85,7 @@ impl EnvVars {
                 plain.insert(k.to_string(), v.to_string());
             }
         }
-        mem::replace(&mut self.plain, plain);
+        self.plain = plain;
```
Comment (janslow) [inline shipcat_definitions/src/structs/env.rs:88]: We're discarding the old values, so there's no need to use `mem::replace`

File: shipcat_definitions/src/vault.rs [hunk 5]
```diff
@@ -10,7 +10,7 @@ fn default_addr() -> Result<String> {
 #[cfg(feature = "filesystem")]
 fn file_token_fallback() -> Result<String> {
     let path = dirs::home_dir()
-        .ok_or_else(|| ErrorKind::NoHomeDirectory)?
+        .ok_or(ErrorKind::NoHomeDirectory)?
```
Comment (janslow) [inline shipcat_definitions/src/vault.rs:13]: unnecessary closure

**Review by nandaloussiBabylon — APPROVED** [review]

**Pull request merged**